app/services/course_potential_subject_service.rb

```diff
@@ -0,0 +1,24 @@
+class CoursePotentialSubjectService
+  def initialize(
+    primary_subject: PrimarySubject,
+    secondary_subject: SecondarySubject,
+    modern_language_subject: ModernLanguagesSubject,
+    further_education_subject: FurtherEducationSubject
+)
+    @primary_subject = primary_subject
+    @secondary_subject = secondary_subject
+    @modern_language_subject = modern_language_subject
+    @further_education_subject = further_education_subject
+  end
+
+  def execute(course)
+    case course.level
+    when 'primary'
+      @primary_subject.all
+    when 'secondary'
+      @secondary_subject.all + @modern_language_subject.all
```

Comment:
i probably break the divide here
if course.subjects has modern language
ModernLanguageSubject .all
else
 @secondary_subject.all